kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableConfig.java

```diff
@@ -198,8 +211,21 @@ private TableConfig(
      */
     public T fromMetadata(Metadata metadata) {
         String value = metadata.getConfiguration().getOrDefault(key, defaultValue);
-        validate(value);
-        return fromString.apply(value);
+        validate(Optional.empty(), value);
+        return fromString.apply(new Tuple2<>(Optional.empty(), value));
+    }
+
+    /**
+     * Returns the value of the table property from the given metadata.
+     *
+     * @param engine {@link Engine} instance.
+     * @param metadata the table metadata
+     * @return the value of the table property
+     */
+    public T fromMetadata(Engine engine, Metadata metadata) {
```

Comment:
I just realized, this is going to be confusing. Lets remove the other `fromMetadata`. Devs need to know if they need the `Engine` or not every time fromMetadata is called.